docs/sqlml.rst

```diff
@@ -107,6 +109,66 @@ the content of "s3://bucket/to/spec.yaml" can be:
 
         YAML-based model spec is still under heavy development.
 
+
+TorchHub Integration
+--------------------
+Rikai supports creating models from the `TorchHub`_ registry. Here is the minimum SQL:
+
+    .. code-block:: sql
+
+        CREATE MODEL resnet50
+        USING "torchhub:///pytorch/vision:v0.9.1/resnet50";
+
+
+It could be expanded to the equivalent and complete SQL:
+
+    .. code-block:: sql
+
+        CREATE MODEL resnet50
+        FLAVOR pytorch
+        OPTIONS (device="cpu")
+        PREPROCESSOR 'rikai.contrib.torchhub.pytorch.vision.resnet50.pre_processing'
+        POSTPROCESSOR 'rikai.contrib.torchhub.pytorch.vision.resnet50.post_processing'
+        RETURNS array<float>
+        USING "torchhub:///pytorch/vision:v0.9.1/resnet50";
+
+
+Most models loaded via TorchHub should adopt the pytorch flavor. In practice, `FLAVOR pytorch`
+can be omitted. Usually, the `FLAVOR` keyword for specifying customized flavor.
+
+TorchHub URI is the only required part.
+    .. code-block:: verbatim
+
+        torchhub:///repo_owner/repo_name[:tag_name]/model_name
+
+
+In this case, here is the corresponding Python snippets to load the model:
+
+    .. code-block:: python
+
+        model = torch.hub.load('pytorch/vision:v0.9.1', 'resnet50', pretrained=True)
+
+
+Given the `repo_owner`, `repo_name` and `model_name`, here is how default PREPROCESSOR and
+POSTPROCESSOR are generated:
+
+    .. code-block:: SQL
+
+        PREPROCESSOR 'rikai.contrib.torchhub.{repo_owner}.{repo_name}.{model_name}.pre_processing'
```

Comment:
what if the model owner in torchhub changes the input/output for a model between two versions / tags?

Reply:
let the user get their hands dirty to handle different version of models

torchhub registery is for exploring

for rikai shipped pre/post processors, there are unit tests to make sure it works with the version shipped by Rikai.

Reply:
What is user story here? My impression of a user journey is that
 

1. User tries the torchhub model over some coco compatible dataset, to visualize coco-defined labels.
2. Train the model using users' own dataset for another purpose (i.e., https://github.com/eto-ai/rikai/issues/456) 
3. Evaluate, debug the torchhub model + user's own dataset, preferably via `ML_PREDICT`
4. Deploy to production

It looks like the `torchhub://` is designed for 1,  how can user do the rest which might be the majority of their workflow?

Reply:
Yes.

`torchhub://` is for new Rikai users to get themselves bootstrapped with minimum dependencies.